fleetshard/pkg/central/operator/upgrade.go

```
@@ -4,67 +4,97 @@ package operator
 import (
 	"context"
 	"fmt"
+	"strings"
 
 	"github.com/golang/glog"
+	apiErrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
 	"github.com/stackrox/acs-fleet-manager/fleetshard/pkg/central/charts"
 	"helm.sh/helm/v3/pkg/chart"
 	"helm.sh/helm/v3/pkg/chartutil"
-	apiErrors "k8s.io/apimachinery/pkg/api/errors"
-	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	ctrlClient "sigs.k8s.io/controller-runtime/pkg/client"
 )
 
 const (
 	operatorNamespace = "stackrox-operator"
 	releaseName       = "rhacs-operator"
-	operatorImage     = "quay.io/rhacs-eng/stackrox-operator:3.74.0"
+	crdKind           = "CustomResourceDefinition"
 )
 
+// chartValue represents operator image
+type chartValue struct {
+	repository string
+	tags       []string
+}
+
+func parseRepositoryAndTags(images []string) (*chartValue, error) {
+	val := chartValue{}
+	for _, img := range images {
+		if strings.Contains(img, ":") {
+			s := strings.Split(img, ":")
+			val.repository = s[0]
+			val.tags = append(val.tags, s[1])
+		} else {
+			glog.Errorf("failed to parse image %s", img)
```

Shouldn't this cause an error to be returned?

---

I am not sure, I think it's better to keep parsing instead of canceling whole process

---

I don't think we should be silently discarding malformed image specs. Better to fail early and loud (but perhaps collect all errors from the parsing and return them together).